### PR TITLE
Feature/trn 751/rewrite transformer

### PR DESCRIPTION
**Pull request by krampstudio**
Related to https://oat-sa.atlassian.net/browse/TRN-751

Rewrite the jest svelte transformer to : 
 - remove `deasync` (that hang with some OS and node versions): use a `child_process.execSync` to communicate synchronously the async preprocess
 - prepare the support of ESM in jest (starting from jest 27) by add the `processAsync` method to the transformer
 - support dynamic import

So this allows to have svelte tests, with dynamic import, code coverage and run with a recent version of node.

How to test:
 - with https://github.com/oat-sa/live-design-system/pull/521 or try to setup the transformer in another repo.

### CODE REVIEW

**Review by olga-kulish — APPROVED** [review]
Works well with LDS. As for the code, I don't have much knowledge in this area.

- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Pull request title and description are meaningful

File: index.js [hunk 1]
```diff
@@ -1,58 +1,158 @@
 const svelte = require('svelte/compiler');
-const deasync = require('deasync');
+const { execSync } = require('child_process');
+const { transformSync, transform } = require('@babel/core');
 
 const styleRegex = /<style[^>]*>[\S\s]*?<\/style>/g;
+const dynImportRegex = /import\(/g;
```
Comment (oatymart) [inline index.js]:
```suggestion
const dynImportRegex = /\bimport\(/g;
```
Maybe with word boundary to *guarantee* that you only match the reserved `import` keyword in JS?

File: index.js [hunk 2]
```diff
@@ -1,58 +1,158 @@
...
+            return svelte.compile(
+                source,
+                Object.assign(
+                    {
+                        filename,
+                        css: false,
+                        // Allow tests to set component props.
+                        accessors: true,
+                        // Debugging and runtime checks
+                        dev: false,
```
Comment (oatymart) [inline index.js:51]: Did you change `true` -> `false` for a reason?

**Review by oatymart — COMMENTED** [review]
I don't know much about the domain so I don't have much to add. ~~Waiting to see if some change will be needed here based on the [test failures I got in Node 14](https://github.com/oat-sa/live-design-system/pull/521#issuecomment-852880728).~~

File: index.js (hunk 2 above)
Comment (krampstudio) [reply index.js:51]: We set `dev: false` in all our projects using the compiler options. It seems to be a good default.

**Review by oatymart — APPROVED** [review]
It looks good and runs our tests.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Documentation is updated according to changes (if applicable)

**Pull request merged**